### DISCUSSION TITLE
Mopping preference / moppingType /sweepType

### DISCUSSION THREAD

**Discussion by ghost** [mrbungle64/ecovacs-deebot.js#279] [Ideas]
Hi,

I saw they handleWaterInfo function receives the sweepType. Any way to *set* the sweepType?

And while were at it, any way to enable/disable the Air Freshning?

:)

Cheers,
/cacti

**Comment by mrbungle64** [comment]
@cactusss 

> I saw they handleWaterInfo function receives the sweepType. Any way to _set_ the sweepType?

No, I haven't checked it yet ...

> 
> And while were at it, any way to enable/disable the Air Freshning?

The same applies here, only with the difference that I don't own a model with air freshener 😉

**Reply by ghost** [reply]
@mrbungle64 

No worries. If you ever get borred and fell like implementing it let me know. I'll be happy to test any additions/changes!

**Comment by mrbungle64** [comment]
@cactusss 

Starting with 0.9.2-alpha.2 you can set the `sweepType`, but currently only with the `SetWaterLevel` command. I don't know if there's a command that exclusively can set the `sweepType` ...

```js
vacbot.run("SetWaterLevel", waterLevel, sweepType);
```

**Reply by mrbungle64** [reply]
@cactusss 

Have you already tried it?

**Comment by ghost** [comment]
@mrbungle64 

Just tried, and it seems to work. Great work Sascha thanks!

Since the mopping type can be set by SetWaterLevel, can’t you make so GetWaterLevel cause an onevent for SweepType? A GetSweepType would also suffice :)

**Reply by mrbungle64** [reply]
@cactusss 

> Since the mopping type can be set by SetWaterLevel, can’t you make so GetWaterLevel cause an onevent for SweepType? A GetSweepType would also suffice :)

There's already an event called `WaterBoxScrubbingType` (and also `WaterBoxMoppingType`).
Is that what you mean?

**Comment by ghost** [comment]
@mrbungle64 

Excellent! FYI, the event WaterBoxScrubbingType work flawlessly, The event WaterBoxMoppingType however does gets fired, but always return with a value 0.

**Comment by mrbungle64** [comment]
@cactusss 

> The event WaterBoxMoppingType however does gets fired, but always return with a value 0.

I think that this is because the bot is in sleeping mode.
Does it return a valid value when the bot is cleaning?

But I'll also have to check the value before firing the event.

**Reply by ghost** [reply]
No, while cleaning the WaterBoxMoppingType isn't fired at all, it only seems to fire when the Deebot is sleeping. While cleaning only the WaterBoxScrubbingType fires, and still give the correct value. So I'll just ignore the WaterBoxMoppingType.

**Reply by mrbungle64** [reply]
@cactusss 

> No, while cleaning the WaterBoxMoppingType isn't fired at all, it only seems to fire when the Deebot is sleeping. While cleaning only the WaterBoxScrubbingType fires, and still give the correct value. So I'll just ignore the WaterBoxMoppingType.

This should be fixed in 0.9.2-alpha.4

**Reply by ghost** [reply]
Thanks. Although I happily use the WaterBoxScrubbingType now. What's the difference between the two, you know?

**Reply by mrbungle64** [reply]
WaterBoxMoppingType (mopping type):
* `1` = Regular
* `2` = OZMO Pro

WaterBoxScrubbingType (scrubbing pattern):
* `1` = Quick scrubbing
* `2` = Deep scrubbing

**Reply by ghost** [reply]
👍🏻
